src/internal/_xtake.js

```diff
@@ -14,8 +14,16 @@ module.exports = (function() {
         return this.xf.result(result);
     };
     XTake.prototype.step = function(result, input) {
-        this.n -= 1;
-        return this.n >= 0 ? this.xf.step(result, input) : _reduced(result);
+        switch (this.n) {
+            case 0:
+                return _reduced(result);
```

Comment:
Is this case necessary, @kevinbeaty? Will the transducer machinery ever invoke `step` if `this.n` is zero?